wormhole-connect/src/views/v2/Redeem/index.tsx

```diff
@@ -164,30 +170,101 @@ const Redeem = () => {
     eta = 0,
   } = txData!;
 
-  const { classes } = useStyles({ transitionDuration: `${eta}ms` });
-
   const getUSDAmount = useUSDamountGetter();
 
+  const details = getTransferDetails(
+    routeName!,
+    tokenKey,
+    receivedTokenKey,
+    fromChain,
+    toChain,
+    amount,
+    getUSDAmount,
+  );
+
+  const { classes } = useStyles({ transitionDuration: `${eta}ms` });
+
+  // Handle changes to receiptState as well as uncaught errors when initiating manual redeems
+  // There are four cases this handles, in this order:
+  //
+  // - receipt.state === DestinationFinalized
+  // - receipt.state === Refunded
+  // - receipt.state === Failed
+  // - Unhandled error when manually redeeming
+  //
+  // Because the unhandled manual redeem error is at the end, we ignore it if
+  // we already saw one of the first three cases.
   useEffect(() => {
-    // When we see the transfer was complete for the first time,
-    // fire a transfer.success telemetry event.
-    if (isTxComplete && !transferSuccessEventFired) {
-      setTransferSuccessEventFired(true);
+    const { receipt } = routeContext;
+
+    if (!receipt) return;
+
+    if (isTxComplete) {
```

Comment:
We should add `isTxComplete`, `transferSuccessEventFired`, `isAutomaticRoute`, and `isTxRefunded` to this hook's deps list

Reply:
We should probably enable the linter rule for exhaustive deps in a different PR